test/mini_racer_test.rb

```diff
@@ -986,33 +999,56 @@ def test_promise
     assert_equal(v, 99)
   end
 
-  def test_webassembly
-    skip "TruffleRuby does not enable WebAssembly by default" if RUBY_ENGINE == "truffleruby"
-    context = MiniRacer::Context.new()
-    context.eval("let instance = null;")
-    filename = File.expand_path("../support/add.wasm", __FILE__)
-    context.attach("loadwasm", proc {|f| File.read(filename).each_byte.to_a})
-    context.attach("print", proc {|f| puts f})
-
-    context.eval <<~JS
-      WebAssembly
-        .instantiate(new Uint8Array(loadwasm()), {
-          wasi_snapshot_preview1: {
-            proc_exit: function() { print("exit"); },
-            args_get: function() { return 0 },
-            args_sizes_get: function() { return 0 }
-          }
-        })
-        .then(i => { instance = i["instance"];})
-        .catch(e => print(e.toString()));
-    JS
-
-    while !context.eval("instance") do
-      context.isolate.pump_message_loop
-    end
-
-    assert_equal(3, context.eval("instance.exports.add(1,2)"))
-  end
+# Fatal error in HandleScope::HandleScope
+# Entering the V8 API without proper locking in place
+# log has been seen as (in order):
+# abcdcdcdcdcdcdcdcdcdcdcdcdc
+# abcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdc
+# abcdcdcdcdcdcdcdcdcdcdcdcdc
+# abcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdc
+# abcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdcdc
+# cuts after c + cuts before d + never e => likely to be Isolate#pump_message_loop
```

Comment:
This one seems to happen randomly after a few loops. Exact cause is yet unknown. Could be some timing, could be message loop doing no work but had some before, could be message loop having some work and none before, could be any other unknown thing really since I did not dig.

Reply:
From: https://github.com/rubyjs/mini_racer/pull/283#issuecomment-1572735274 this was fixed by @Fayti1703 in https://github.com/Fayti1703/mini_racer/commit/1e43a49ae550f8722071c92ca1248000ab49d6f3, which I've cherry-picked.